### ISSUE TITLE
As GM/AD I shouldn't be able to add a user who didn't complete the registration process to a group I edit/create PASSBOLT-2078 

### ISSUE THREAD

**Issue by ktosiek** [passbolt/passbolt_api#75]
Server version: 1.5.1
Plugin version: 1.5.1rc2 (Firefox)

Adding new (added, but not fully registered) users to a group fails. They can be selected from dropdown in group editor, but after selecting them I'm getting a JS error:

```
TypeError: mappedItem.userFingerprint is null[Więcej informacji] passbolt.dcs.pl:505:324
	<anonimowa> https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:505:324
	n.prototype.render https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:143:462
	n/< https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:143:258
	i/r.render https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:60:200
	renderTo https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:60:3082
	renderAs https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:60:3722
	render https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:60:3013
	render https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:141:320
	insertItem https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:189:1085
	insertItem https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:195:1422
	loadGroupUser https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:507:3317
	addGroupUser https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:507:3804
	{mad.bus.element} passbolt.group.edit.add_user https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:507:6283
	_shifter/< https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:44:724
	dispatch https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:14488
	add/m.handle https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:11298
	trigger https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:13648
	trigger/< https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:19614
	each https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:6:12111
	each https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:6:10085
	trigger https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:19593
	<anonimowa> https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:167:687
	passbolt.message.emitToPage resource://passbolt-at-passbolt-dot-com/data/js/lib/message.js:93:5
	<anonimowa> resource://passbolt-at-passbolt-dot-com/data/js/app.js:105:5
	Deferred/then/</</< resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:28467
	i resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:27146
	fireWith resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:27914
	handleResponse resource://passbolt-at-passbolt-dot-com/data/js/lib/request.js:103:9
	onceCallback resource://gre/modules/commonjs/sdk/content/content-worker.js:27:11
	onEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:45:22
	onEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:45:22
	onChromeEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:91:16
	emitToContent resource://gre/modules/commonjs/sdk/content/sandbox.js:411:10
	emit/< resource://gre/modules/commonjs/sdk/content/sandbox.js:80:7
	delay/< resource://gre/modules/commonjs/sdk/lang/functional/concurrent.js:38:20
	notify resource://gre/modules/commonjs/sdk/timers.js:40:9

17:49:23.750 TypeError: s.body.Group is undefined[Więcej informacji]  passbolt.dcs.pl:507:7258
	{mad.bus.element} group_edit_save_error https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:507:7258
	_shifter/< https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:44:724
	dispatch https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:14488
	add/m.handle https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:11298
	trigger https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:13648
	trigger/< https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:19614
	each https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:6:12111
	each https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:6:10085
	trigger https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:7:19593
	<anonimowa> https://passbolt.dcs.pl/js/dist/bundles/app/passbolt.js:167:687
	passbolt.message.emitToPage resource://passbolt-at-passbolt-dot-com/data/js/lib/message.js:93:5
	<anonimowa> resource://passbolt-at-passbolt-dot-com/data/js/app.js:162:9
	Deferred/then/</</< resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:28467
	i resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:27146
	fireWith resource://passbolt-at-passbolt-dot-com/data/vendors/jquery.min.js:2:27914
	handleResponse resource://passbolt-at-passbolt-dot-com/data/js/lib/request.js:107:9
	onceCallback resource://gre/modules/commonjs/sdk/content/content-worker.js:27:11
	onEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:45:22
	onEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:45:22
	onChromeEvent resource://gre/modules/commonjs/sdk/content/content-worker.js:91:16
	emitToContent resource://gre/modules/commonjs/sdk/content/sandbox.js:411:10
	emit/< resource://gre/modules/commonjs/sdk/content/sandbox.js:80:7
	delay/< resource://gre/modules/commonjs/sdk/lang/functional/concurrent.js:38:20
	notify resource://gre/modules/commonjs/sdk/timers.js:40:9
```

I've expected to either be able to add them to groups, or get a warning about them not having registered public keys.

**Comment by cedricalfonsi** [comment]
Thanks for reporting!
I've created a ticket in our JIRA (PASSBOLT-2078). We'll fix it soon.

**Comment by cedricalfonsi** [comment]
Fixed with v1.6.0
